test/test_linalg.py

```diff
@@ -31,6 +32,125 @@ def fn(a, b):
+                f'from_dtype={from_dtype}, to_dtype={to_dtype}')
+            input = torch.randn(*input_size, dtype=from_dtype, device=device)
+            result = torch.linalg.norm(input, ord, keepdim=keepdim, dtype=from_dtype)
+            self.assertEqual(result.dtype, from_dtype, msg=msg)
+            result_converted = torch.linalg.norm(input, ord, keepdim=keepdim, dtype=to_dtype)
+            self.assertEqual(result_converted.dtype, to_dtype, msg=msg)
+            self.assertEqual(result.to(compare_dtype), result_converted.to(compare_dtype), msg=msg)
+
+            result_out_converted = torch.empty_like(result_converted)
+            torch.linalg.norm(input, ord, keepdim=keepdim, dtype=to_dtype, out=result_out_converted)
+            self.assertEqual(result_out_converted.dtype, to_dtype, msg=msg)
+            self.assertEqual(result_converted, result_out_converted, msg=msg)
+
+        ord_vector = [0, 1, -1, 2, -2, 3, -3, 4.5, -4.5, inf, -inf, None]
+        ord_matrix = [1, -1, 2, -2, inf, -inf, None]
+        S = 10
+        test_cases = [
+            ((S, ), ord_vector),
+            ((S, S), ord_matrix),
+        ]
+        for keepdim in [True, False]:
+            for input_size, ord_settings in test_cases:
+                for ord in ord_settings:
+                    # float to double
+                    run_test_case(input_size, ord, keepdim, torch.float, torch.double, torch.float)
+                    # double to float
+                    run_test_case(input_size, ord, keepdim, torch.double, torch.double, torch.float)
+
+        # TODO: Once dtype arg is supported in nuclear and frobenius norms, remove the following test
+        #       and  add 'nuc' and 'fro' to ord_matrix above
+        for ord in ['nuc', 'fro']:
+            input = torch.randn(10, 10, device=device)
+            with self.assertRaisesRegex(ValueError, r'dtype argument is currently not supported'):
+                torch.linalg.norm(input, ord, dtype=torch.float)
+
+    @unittest.skipIf(not TEST_NUMPY, "NumPy not found")
+    @dtypes(torch.float, torch.double)
+    def test_norm(self, device, dtype):
+        def run_test_case(input, p, dim, keepdim):
+            result = torch.linalg.norm(input, ord, dim, keepdim)
+            input_numpy = input.cpu().numpy()
+            result_numpy = np.linalg.norm(input_numpy, ord, dim, keepdim)
+
+            msg = f'input.size()={input.size()}, ord={ord}, dim={dim}, keepdim={keepdim}, dtype={dtype}'
+            self.assertEqual(result, result_numpy, msg=msg)
+
+            result_out = torch.empty_like(result)
+            torch.linalg.norm(input, ord, dim, keepdim, out=result_out)
+            self.assertEqual(result, result_out, msg=msg)
+
+        ord_vector = [0, 1, -1, 2, -2, 3, -3, 4.5, -4.5, inf, -inf, None]
+        ord_matrix = [1, -1, 2, -2, inf, -inf, 'nuc', 'fro', None]
+        S = 10
+        test_cases = [
+            # input size, p settings, dim
+
+            # Vector norms
```

Comment:
Nit: for additional test granularity may want to separate vector and matrix norm testing